### PR TITLE
Added implementation for deploy plugin

### PR DESCRIPTION
**Pull request by johnsca**
Had to downgrade PyYAML version due to conflict with jujubundlelib.

### CODE REVIEW

File: matrix/rules.py [hunk 1]
```diff
@@ -303,8 +305,10 @@ def exception_handler(self, context, loop=None, exc_ctx=None):
         reporter = functools.partial(self.exception_handler, context)
         self.loop.set_exception_handler(reporter)
         try:
+            await context.juju_model.connect_current()
```
Comment (bcsaller) [inline matrix/rules.py]: This is fine for now, but I think we'll want to expose controls in the future that let us do a model per test or model per suite.

File: tox.ini [hunk 2]
```diff
@@ -11,7 +11,8 @@ skipsdist = True
 install_command =  pip install {opts} {packages}
 commands = py.test {posargs}
 deps =
-    {toxinidir}
+    -e{toxinidir}
+    --no-index
```
Comment (johnsca) [inline tox.ini]: NB: This is needed to work around juju-solutions/python-libjuju#2.  If it's not included, then re/building the tox env tries to pull in the conflicting juju package due to the higher version number than what's in the wheelhouse.

File: matrix/rules.py (hunk 1 above)
Comment (johnsca) [inline matrix/rules.py]: Agreed.  `libjuju` doesn't currently have that functionality, though.  I had a TODO comment in here but I accidentally dropped it during a refactor.

**Comment by petevg** [comment]
LGTM/+1

**Pull request merged**